CONTRIBUTING.rst

```diff
@@ -0,0 +1,238 @@
+Contribution getting started
+============================
+
+Contributions are highly welcomed and appreciated.  Every little help counts,
+so do not hesitate! If you like tox, also share some love on Twitter or in your blog posts.
+
+.. contents:: Contribution links
+   :depth: 2
+
+.. _submitfeedback:
+
+Feature requests and feedback
+-----------------------------
+
+We'd also like to hear about your propositions and suggestions.  Feel free to
+`submit them as issues <https://github.com/tox-dev/tox/issues>`_ and:
+
+* Explain in detail how they should work.
+* Keep the scope as narrow as possible.  This will make it easier to implement.
+
+.. _reportbugs:
+
+Report bugs
+-----------
+
+Report bugs for tox in the `issue tracker <https://github.com/tox-dev/tox/issues>`_.
+
+If you are reporting a bug, please include:
+
+* Your operating system name and version.
+* Any details about your local setup that might be helpful in troubleshooting,
+  specifically the Python interpreter version, installed libraries, and tox
+  version.
+* Detailed steps to reproduce the bug, or - even better, a n xfaling test reproduces the bug
+
+If you can write a demonstration test that currently fails but should pass
+(xfail), that is a very useful commit to make as well, even if you cannot
+fix the bug itself (e.g. something like this in
+`test_config <https://github.com/tox-dev/tox/blob/2.8.2/tests/test_config.py#L2206>)`_
+
+.. _fixbugs:
+
+Fix bugs
+--------
+
+Look through the GitHub issues for bugs.  Here is a filter you can use:
+https://github.com/tox-dev/tox/labels/bug
+
+Don't forget to check the issue trackers of your favourite plugins, too!
+
+.. _writeplugins:
+
+Implement features
+------------------
+
+Look through the GitHub issues for enhancements.  Here is a filter you can use:
+https://github.com/tox-dev/tox/labels/enhancement
+
+Write documentation
+-------------------
+
+tox could always use more documentation.  What exactly is needed?
+
+* More complementary documentation.  Have you perhaps found something unclear?
+* Docstrings.  There can never be too many of them.
+* Blog posts, articles and such -- they're all very appreciated.
+
+You can also edit documentation files directly in the GitHub web interface,
+without using a local copy.  This can be convenient for small fixes.
+
+.. note::
+    Build the documentation locally with the following command:
+
+    .. code:: bash
+
+        $ tox -e docs
+
+    The built documentation should be available in the ``doc/_build/``.
+
+.. _submitplugin:
+
+.. _`pull requests`:
+.. _pull-requests:
+
+Preparing Pull Requests
+-----------------------
+
+Short version
+^^^^^^^^^^^^^
+
+#. `Fork the repository<https://help.github.com/articles/fork-a-repo/>`_
+#. Make your changes.
+#. open a `pull request <https://help.github.com/articles/about-pull-requests/>`_ targeting the ``master`` branch.
+#. Follow **PEP-8**. There's a ``tox`` command to help fixing it: ``tox -e fix-lint``.
```

Comment:
probably also mention that we have a style check tox environment that warn about non fixes, e.g. flake8